### ISSUE TITLE
Branding: Theia should has favicon icon

### ISSUE THREAD

**Issue by AndrienkoAleksandr** [theia-ide/theia#3600]
Branding: Theia should has favicon icon.

**Comment by vince-fugnitto** [comment]
@kittaakos can you perhaps share on how gitpod achieved this? The only way I could see updating the icon in the menu-bar was by editing the css (`theia-icon`) from core.

**Comment by kittaakos** [comment]
> can you perhaps share on how gitpod achieved this

I do not know.

**Comment by vince-fugnitto** [comment]
I was able to test and add a custom `favicon` to the main application by updating 
the `frontend-generator.ts`. Where we generate the `<head></head>` portion, I included the following link tag.

```ts
  <link rel="icon" href={url} type="image/x-icon">
```

The only issue I had was getting the image served by the server. When using an external image (such as one hosted already) the icon was displayed correctly.

<div align='center'>

![Screenshot from 2019-03-12 08-48-20](https://user-images.githubusercontent.com/40359487/54201299-aaf43580-44a3-11e9-8daa-3b82747f9aa6.png)

</div>

The next step would be to find out how to properly serve internal icons as a `favicon` and make it much easier for extenders to update it.

**Comment by ektek** [comment]
Thank you for opening this issue. Has there been any update on making it easy for extensions to set a favicon?

**Comment by scheng123** [comment]
For anyone that is desperate for a favicon, here's a little command you can run to add a favicon to your theia web ide:
```
cd your_theia_folder/lib && wget https://raw.githubusercontent.com/theia-ide/theia-apps/master/theia-electron/resources/icon.ico && sed -i 's/<meta charset/<link rel="icon" href=".\/icon.ico" type="image\/x-icon">\n  <meta charset=/' index.html 
```

**Comment by krishnagogada** [comment]
It work for me in this method `preloadTemplate`. 
In package.json
```
"theia": {
    "generator": {
      "config": {
        "preloadTemplate": "preloadTemplate.html"
      }
    }
}
```
in `preloadTemplate.html`

```
<div>Loading view.......</div>
<script>
if (document.head) {
        let faviconLinkTag = document.createElement('link');
        faviconLinkTag.rel = 'icon';
        faviconLinkTag.href = 'http://your_favicon.ico';
        document.head.appendChild(faviconLinkTag);
    }
</script>
```

**Comment by JonasHelming** [comment]
@vince-fugnitto Do you have  a preferred way to add a favicon to a custom product? If so, we should document it.

**Comment by JonasHelming** [comment]
Yes, that makes sense. I was more interest in your view, which of the above described options to customize it you prefer as the officially documented one?

**Comment by capsulated** [comment]
Hello everyone! Any news about favicon?

**Comment by msujew** [comment]
@capsulated This topic has been effectively resolved in [this comment](https://github.com/eclipse-theia/theia/issues/3600#issuecomment-808747788).

**Comment by hkuich** [comment]
> It work for me in this method `preloadTemplate`. In package.json
> 
> ```
> "theia": {
>     "generator": {
>       "config": {
>         "preloadTemplate": "preloadTemplate.html"
>       }
>     }
> }
> ```
> 
> in `preloadTemplate.html`
> 
> ```
> <div>Loading view.......</div>
> <script>
> if (document.head) {
>         let faviconLinkTag = document.createElement('link');
>         faviconLinkTag.rel = 'icon';
>         faviconLinkTag.href = 'http://your_favicon.ico';
>         document.head.appendChild(faviconLinkTag);
>     }
> </script>
> ```

Just to make sure: there is supposedly more in the preloadTemplate.html? I can't seem to get this to work... Would be very thankful for a helping hand for a Theia novice...

**Comment by capsulated** [comment]
> @capsulated This topic has been effectively resolved in [this comment](https://github.com/eclipse-theia/theia/issues/3600#issuecomment-808747788).

@msujew Thanks, that is works!
